File: TheRayCode/Java/Behavioral/Interpreter/Example/Interpreter/src/SubtractExpression.java
public class SubtractExpression implements Expression {
    private Expression firstExpression;
    private Expression secondExpression;

    public SubtractExpression(Expression firstExpression, Expression secondExpression) {
        this.firstExpression = firstExpression;
        this.secondExpression = secondExpression;
    }

    @Override
    public int interpret() {
        return this.firstExpression.interpret() - this.secondExpression.interpret();
    }
}

File: TheRayCode/CSharp/Behavioral/Interpreter/Example/Interpreter/Interpreter/SubtractExpression.cs
namespace Interpreter
{
    public class SubtractExpression: IExpression
    {
        private readonly IExpression _leftExpression;
        private readonly IExpression _rightExpression;

        public SubtractExpression(IExpression left, IExpression right)
        {
            _leftExpression = left;
            _rightExpression = right;
        }

        public int Interpret()
        {
            return _leftExpression.Interpret() - _rightExpression.Interpret();
        }
    }
}